src/app/codegen-data-model/CodegenDataModel_Write.cpp

```diff
@@ -0,0 +1,381 @@
+
+        VerifyOrReturnError(!(*attributeMetadata)->MustUseTimedWrite() ||
+                                request.writeFlags.Has(InteractionModel::WriteFlags::kTimed),
+                            CHIP_IM_GLOBAL_STATUS(NeedsTimedInteraction));
+    }
+
+    // Extra check: internal requests can bypass the read only check, however global attributes
+    // have no underlying storage, so write still cannot be done
+    VerifyOrReturnError(attributeMetadata != nullptr, CHIP_IM_GLOBAL_STATUS(UnsupportedWrite));
+
+    if (request.path.mDataVersion.HasValue())
+    {
+        std::optional<InteractionModel::ClusterInfo> clusterInfo = GetClusterInfo(request.path);
+        if (!clusterInfo.has_value())
+        {
+            ChipLogError(DataManagement, "Unable to get cluster info for Endpoint 0x%x, Cluster " ChipLogFormatMEI,
+                         request.path.mEndpointId, ChipLogValueMEI(request.path.mClusterId));
+            return CHIP_IM_GLOBAL_STATUS(DataVersionMismatch);
+        }
+
+        if (request.path.mDataVersion.Value() != clusterInfo->dataVersion)
+        {
+            ChipLogError(DataManagement, "Write Version mismatch for Endpoint 0x%x, Cluster " ChipLogFormatMEI,
+                         request.path.mEndpointId, ChipLogValueMEI(request.path.mClusterId));
+            return CHIP_IM_GLOBAL_STATUS(DataVersionMismatch);
+        }
+    }
+
+    AttributeAccessInterface * aai       = GetAttributeAccessOverride(request.path.mEndpointId, request.path.mClusterId);
+    std::optional<CHIP_ERROR> aai_result = TryWriteViaAccessInterface(request.path, aai, decoder);
+    if (aai_result.has_value())
+    {
+        return *aai_result;
+    }
+
+    ReturnErrorOnFailure(DecodeValueIntoEmberBuffer(decoder, *attributeMetadata, gEmberAttributeIOBufferSpan));
+
+    Protocols::InteractionModel::Status status;
+
+    if (!request.operationFlags.Has(InteractionModel::OperationFlags::kInternal))
+    {
+
+        EmberAfAttributeSearchRecord record;
+        record.endpoint    = request.path.mEndpointId;
+        record.clusterId   = request.path.mClusterId;
+        record.attributeId = request.path.mAttributeId;
+        // NOTE: gEmberAttributeIOBufferSpah is likely much larger than the attribute itself. Ember is supposed
+        //       to be able to pick up the "valid size" out of this
+        //
+        // Specifically `When writing attributes, readLength is ignored.`.
+        // The actual length is used by unit tests to copy over data without needing "get attribute size" logic
+        // since metadata attribute sizes in our mocks is not generally correct.
+        status = emAfReadOrWriteAttribute(&record, attributeMetadata, /* buffer = */ gEmberAttributeIOBufferSpan.data(),
+                                          /* readLength = */ static_cast<uint16_t>(gEmberAttributeIOBufferSpan.size()),
+                                          /* write = */ true);
```

Comment:
OK, so this "internal" write is, again:

1. Skipping notifying that the value changed, for subscription reporting.  Is that expected?
2. Skipping pre/post write callbacks (maybe OK).
3. Skipping NVM handling (probably not ok).

Reply:
My goal is to have the ability to have the Accessors.cpp equivalent. Updated to use emberAfWriteAttribute since that seems used there.

I did not even check at the beginning as this seemed awfully asymmetric - the read uses `emAfReadOrWriteAttribute` in compatibility functions, where as the writes seem to have different wrappers. Accessors.cpp seems to use `emberAfReadAttribute` for reading, however that seems to be just a thin wrapper around emAfReadOrWriteAttribute + an extra log for resource exhausted.